src/go/rpk/pkg/cli/cmd/container/start_test.go

```diff
@@ -560,3 +586,115 @@ func TestCollectFlags(t *testing.T) {
 		})
 	}
 }
+
+func TestAssignPorts(t *testing.T) {
+	tests := []struct {
+		name           string
+		ports          []string
+		nodesCount     uint
+		portName       string
+		expectedError  string
+		expectedPorts  []uint
+		isAnySpecified bool
+	}{
+		{
+			name:          "it should not allow if ports is greater than the number of nodes",
```

Comment:
[nit/suggestion] we can keep the names short, we can drop the 'it should' because is implicit and repeated in every test name

Also, we can shorten the names as:

```
		name           string
		ports          []string
		nodes          uint
		portName       string  # <-- Can we drop this one? is always the same, we can just hardcode it in the test
		expErr         string
		exp            []uint
		isAny          bool
```

You can take this as an example: https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/config/weak_test.go#L10